esmvalcore/_recipe.py

```diff
@@ -890,6 +967,36 @@ def _get_preprocessor_task(variables, profiles, config_user, task_name):
             )
             derive_tasks.append(task)
 
+    # special case: fx variable pre-processing
+    for step in ('area_statistics', 'volume_statistics', 'zonal_statistics'):
+        if profile.get(step, {}).get('fx_files') and \
+                variable.get("fx_var_preprocess"):
+            # conserve profile
+            fx_profile = deepcopy(profile)
+            fx_vars = profile.get(step, {}).get('fx_files')
+
+            # Create tasks to prepare the input data for the fx var
+            order = _extract_preprocessor_order(fx_profile)
+            before, _ = _split_settings(fx_profile, step, order)
+            for var in variables:
+                fx_variables = [
+                    _get_correct_fx_file(var, fx_var, config_user)[1]
+                    for fx_var in fx_vars
+                ]
+                for fx_variable in fx_variables:
+                    fx_name = task_name.split(
```

Comment:
I think we need some kind of raise here for when the fx file is not found. 

At the moment this is flagged as a list index problem. If no fx files are found, then `fx_variable` is a empty list and `fx_variable['variable_group']` on line 996 just raises the error:
```
  File "/users/modellers/ledm/workspace/ESMValTool_AR6/ESMValCore/esmvalcore/_recipe.py", line 996, in _get_preprocessor_task
    fx_variable['variable_group']
TypeError: list indices must be integers or slices, not str


```

Reply:
This error can be reproduced with the recipe:

```
# ESMValTool
# test_439.yml
---
documentation:
  description: |
    Recipe to test PR 439

  authors:
    - demora_lee

  projects:
    - ukesm

preprocessors:
    prep_sst:
        custom_order: True
        annual_statistics:
            operator: mean
        extract_region:
            start_longitude: -80.
            end_longitude: 30.
            start_latitude: -80.
            end_latitude: 80.
        area_statistics:
            operator: mean
            fx_files: [{short_name: areacello, mip: Ofx, exp: piControl}]
        regrid_time:
            frequency: yr

diagnostics:
  diag_439:
      variables:
        thetaosst:
          short_name: tos
          preprocessor: prep_sst
          fx_var_preprocess: True
          additional_datasets:
            - {dataset: GFDL-CM4,        project: CMIP6, mip: Omon, exp: historical, ensemble: r1i1p1f1, grid: gn, start_year: 1960, end_year: 1963}

      scripts:
        diagnostic_439:
          script: ocean/diagnostic_timeseries.py
          #script: ocean/diagnostic_fig_3_20_OHC_noppderive_timeseries.py

```

Reply:
Note that this recipe will always fail, as there is no `GFDL-CM4 areacello` data available o  jasmin.